python/graphstorm/distiller/distiller.py

```diff
@@ -0,0 +1,553 @@
+        world_size : int
+            Number of workers.
+
+        Returns
+        -------
+        list : list for the file names for the local worker.
+        """
+        num_files = len(total_file_list)
+        part_len = num_files // world_size
+        remainder = num_files % world_size
+        part_start = part_len * local_rank + min(local_rank, remainder)
+        part_end = part_start + part_len + (local_rank < remainder)
+        part_len = part_end - part_start
+
+        if part_len == 0:
+            return []
+
+        local_file_list = []
+        for i in range(part_len):
+            local_file_list.append(total_file_list[part_start+i])
+        return local_file_list
+
+    def map_text_embeds(
+        self,
+        queue,
+        node_type,
+        file_name,
+        id_field,
+        textual_field_list,
+        disk_embeds,
+        id_map,
+        index,
+        total_num,
+        concat_field_name=False,
+        split="train",
+        chunk_size=10000,
+        part=0,
+        rank=0,
+    ): 
+        """ Map textual features with embeddings for each file.
+        Embeddings are read from HDF5Array on disk,
+        so no embeddings copy for each worker. A queue is maintained
+        for each worker to store the samples. Joined files will be written
+        to disk when queue size hits the chunk size. This is to make sure
+        the same batch numbers during training of distillation.
+
+        Parameters
+        ----------
+        queue: dict
+            Store joined samples to be written to disk.
+        node_type : str
+            node type.
+        file_name : str
+            Name of the textual file.
+        id_field : str
+            Name of string ID field in textual data.
+        textual_field_list : list of str
+            List of textual field name to be considered.
+        disk_embeds : HDF5Array
+            Offset to embeddings map.
+        id_map : dict
+            String ID to offset map.
+        index : int
+            Index of current textual file
+        total_num : int
+            Number of local textual files
+        concat_field_name : bool
+            Whether to concat field name with textual features.
+        split : str
+            Split of current textual file.
+        chunk_size : int
+            Chunk size for the output joined file when writing to disk.
+        part : int
+            Partition index of the output joined file.
+        rank : int
+            Local rank.
+
+
+        Returns
+        -------
+        queue: dict
+            Store joined samples to be written to disk.
+        part : int
+            Partition index of the output joined file.
+        """
+        print ("Worker {}: Map {} textual features: {} of {}".format(self.rank, node_type, index, total_num))
+        assert len(queue["node_ids"]) < chunk_size, "Queue already greater than chunk size"
+
+        embed_offset = []
+
+        with open(file_name, "r") as reader:
```

Comment:
what does the file store? it's a json file? i thought you store the text data in parquet file.

Reply:
this is the textual data provided by user in json format.